sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/SymbolicExecution/InvocationVisitor.cs

```diff
@@ -160,18 +184,21 @@ private ProgramState HandleInstanceEqualsCall()
                 .PopValues(validatedArgumentIndex)
                 .SetConstraint(guardedArgumentValue, ObjectConstraint.NotNull);
 
-        private static readonly ISet<string> IsNullMethodNames = new HashSet<string>
-        {
-            nameof(string.IsNullOrEmpty),
-            nameof(string.IsNullOrWhiteSpace),
-        };
 
         private static bool IsStringNullCheckMethod(IMethodSymbol methodSymbol)
         {
             return methodSymbol != null &&
                 methodSymbol.ContainingType.Is(KnownType.System_String) &&
                 methodSymbol.IsStatic &&
-                IsNullMethodNames.Contains(methodSymbol.Name);
+                nameof(string.IsNullOrWhiteSpace) == methodSymbol.Name;

```

Comment:
should the name of this method be `IsStringNullOrWhiteSpaceCheckMethod` now?

well, it would be quite confusing (NullOrEmpty and NullOrWhitespace)... then add a comment to say why we treat them differently?